PDF Upload, Zube 90: Shrine, Uppy, and S3 Configuration

tinadev opened the pull request:
## Summary
Updated Rails to 5.2.3 🎉 . Also added `webpacker` 🐩 's 🎉 🎉 🎊 . Boom. 🎤 drop. 

Generated a migration on `locations` to add a column for `file_uploads`. Added that association to the location model (a location has many file uploads). Updated the location controller to permit the file uploads attributes. 

Added Uppy and Shrine to the project for the file uploads. 

**Zube Card Referenced** 90

**File Addition(s)** 
`application/javascript/packs/application.js`: The entry point for webpack
`application/javascript/fileUpload`: The configuration for uppy.

**Files Modified**
- `location.rb`: Added 'English' to the list of languages

**Libraries Added**
- `gem 'webpacker', '~> 5.1'`: Used to add webpack
- `yarn add uppy`

### Migrations to Run: Yes (adds column to locations) 

### Tests to Run: TODO
- `spec/features/whatver_spec.rb`: Tests the function `hello_world`

### TODO
- [ ] Commit test for file upload

Commit:
Shrine, Uppy, and S3 Configuration

+Added shrine and uppy

Commit:
Config: Application Upload Server

+Changed upload to multipart instead of s3

Commit:
File Upload: Form and Migration

+AddColumnFileUpload migration for location
-Each location has many files
+Added the file upload form templates and rendering
+Added the uppy js file (will be moved)

Commit:
Rails Update and Webpack

+Updated rails to 5.2.3
+Added webpacker

justuseapen commented on app/models/file_upload.rb:
```diff
@@ -0,0 +1,5 @@
+class FileUpload < ActiveRecord::Base
+  include ImageUploader::Attachment(:image)
```
Is a pdf considered an image? Has this been tested with pdfs?

tinadev replied:
Yes and yes

Commit:
Gitignore

+Added public uplaod cache to ignore:wq

Commit:
Package

+Added randomstring

Commit:
Routes: Formatted

Commit:
Merge remote-tracking branch 'origin' into locations-pdf-upload-90

Commit:
File Upload: Migration change column name

+Change file_data to image_data

Commit:
File Upload: Fix form not saving file

+Locations Controller: add line to assign attributes
+File Upload Helper: Created helper to fetch the correct route depending on environment
+File Upload JS: Added hidden field name for saving the image data for file
+File Upload Partial: Renders table of files saved to location
+File Upload Form: Renders table for file upload partial

Commit:
Location Serializer

+Open file uploads for UI

Commit:
Merge branch 'develop' into locations-pdf-upload-90

danivovich commented on db/structure.sql:
```diff
@@ -1541,5 +1812,7 @@ INSERT INTO "schema_migrations" (version) VALUES
 ('20200621214426'),
 ('20200621222659'),
 ('20200621223318');
+('20200629173821'),
```
This is a bad merge it seems, the semicolon in the line above will not let these 2 additional versions be valid

tinadev replied:
Fixed. Thank you

Commit:
Structure: Fixed syntax error

Commit:
Merge branch 'locations-pdf-upload-90' of https://github.com/bchd/bahc-ohana-api into locations-pdf-upload-90

Commit:
Structure: Cleanup

+Removed inaccurate_resources and inaccurate_resource_categories that were pushed up from local
-Created migrations to drop those tables
-Migration are designed to be irreversible

Commit:
Drone YML: Webapcker

+Added step for compiling webpack

Commit:
Drone: Hereee we goo..

+Going to tinker this until CI passes

Commit:
drone

Commit:
More drone

Commit:
Drone: Reconfigure to fix webpack error

Commit:
Drone: Installing yarn

Commit:
Drone: Fix Yarn error

Commit:
Drone: Fix sudo error

Commit:
Drone: Database migration fix

Commit:
Drone: Delete conflicting migrations

Commit:
Drone: Config update

Commit:
Drone: Chnage node version to 12

Commit:
Drone: Postgres Version

Commit:
Drone: Change lsb release to buster

Commit:
Drone: Stretch instead of buster

Commit:
Merge branch 'develop' of https://github.com/bchd/bahc-ohana-api into locations-pdf-upload-90

Commit:
Structure SQL

+Drop db and reset

Commit:
Development and Test: Configure Mailure

+Add action mailure default url for tests

Commit:
Merge branch 'develop' of https://github.com/bchd/bahc-ohana-api into locations-pdf-upload-90

donaldguy committed:
Remove redundant js inclusion; pin yarn and nodejs versions (dev)

donaldguy committed:
specs: output in color even in drone

donaldguy committed:
do verbose rspec output in ci

donaldguy committed:
Adapt Admin js (select2 and sdme) init to turbolinks

donaldguy committed:
Revert select2 upgrade until I can update test helpers

donaldguy committed:
Revert "Revert select2 upgrade until I can update test helpers"

This reverts commit 9b4273609f7d28c168520c8069cbc48e43a6243a.

donaldguy committed:
Use non select2-specific testing logic for "#service_funding_sources"

donaldguy committed:
Use non select2-specific testing logic for "#service_languages"

donaldguy committed:
Use non select2-specific testing logic for "#service_required_documents"

donaldguy committed:
Use non select2-specific testing logic for "#service_service_areas"

hidden input is still pushing an empty string here ... but not on required_documents... for some reason?

donaldguy committed:
Use non select2-specific testing logic for "#service_accepted_payments"

donaldguy committed:
Use non select2-specific testing logic for "#organization_funding_sources"

donaldguy committed:
Upgrade "#organization_licenseses" tag select2 to use select rather than input

See also:  https://select2.org/upgrading/migrating-from-35#no-more-hidden-input-tags

donaldguy committed:
Upgrade "#organization_accreditations" tag select2 to use select rather than input

See also:  https://select2.org/upgrading/migrating-from-35#no-more-hidden-input-tags

donaldguy committed:
Upgrade "#service_keywords" tag select2 to use select rather than input

See also:  https://select2.org/upgrading/migrating-from-35#no-more-hidden-input-tags

donaldguy committed:
Adapt #org-name interaction to upgraded select2

https://select2.org/upgrading/migrating-from-35#no-more-hidden-input-tags

donaldguy committed:
Select2 on location-language-select, formerly location_languages

donaldguy committed:
Remove select2 helpers

donaldguy committed:
Remove extra console.logs

donaldguy committed:
Remove hanging reference to deleted preprocess_organization_params

donaldguy committed:
Missed language update case

donaldguy committed:
fix a few more tests

donaldguy committed:
Merge pull request #129 from bchd/donald-select2-reupgrade

(Re)upgrade Select2 for #112

donaldguy committed:
Explicitly dedup service arrays as dump to yaml

Did auto_strip_attributes used to do this

donaldguy committed:
Add pg_array validation to location.admin_emails; this dedups for some reason

Did auto_strip_attributes used to do this?

25 failures -> 20 failures

donaldguy committed:
Chromedriver release has gotten ahead of chrome deb releases

donaldguy committed:
Get to 19 tests?

- remove a leftover pry
- don't let ruby bootstrap its own chromedriver badly
- surpress locatio.admin_emails: [] -> nil

donaldguy committed:
17 failures. use Figaro sourced action_mailer url name

donaldguy committed:
Fix some more tests

donaldguy committed:
Remove 2 tests verifying that routes that don't exist don't exist

ActionDispatch apparently now (correctly) complains about that
(and raises an exception)

donaldguy committed:
Don't transit a double assoc if first one null (might be a test only concern?)

donaldguy committed:
Avoid phone number double insert

After update_attributes added in 001902ff66504fbca16dc6822fe12b0b173cdc38

We only need a save, not a full update call

donaldguy committed:
Remove html-ization of description that was always a lie?

donaldguy committed:
Merge remote-tracking branch 'origin/develop' into 112-rest-of-the-tests

Commit:
Merge branch 'develop' of https://github.com/bchd/bahc-ohana-api into locations-pdf-upload-90

Commit:
:hammer_and_wrench: Gemfile Lock

+Deleted lock and ran bundle install
-Hopefully this fixes conflicts with
develop

Commit:
Merge branch 'develop' into locations-pdf-upload-90

donaldguy committed:
Merge branch '112-rest-of-the-tests' into locations-pdf-upload-90
